src/assets/rewardrequestdb.cpp

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) 2019 The Raven Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <set>
+
+#include "rewardrequestdb.h"
+
+static const char SCHEDULEDREWARD_FLAG = 'S';
+
+CRewardRequest::CRewardRequest()
+{
+    SetNull();
+}
+
+CRewardRequest::CRewardRequest(
+    const std::string & p_rewardID,
+    const std::string & p_walletName, const int p_heightForPayout, const CAmount p_totalPayoutAmt,
+    const std::string & p_payoutSrc, const std::string & p_tgtAssetName, const std::string & p_exceptionAddresses
+)
+{
+    SetNull();
+
+    rewardID = p_rewardID;
+    walletName = p_walletName;
+    heightForPayout = p_heightForPayout;
+    totalPayoutAmt = p_totalPayoutAmt;
+    payoutSrc = p_payoutSrc;
+    tgtAssetName = p_tgtAssetName;
+    exceptionAddresses = p_exceptionAddresses;
+}
+
+CRewardRequestDBEntry::CRewardRequestDBEntry()
+{
+    SetNull();
+}
+
+CRewardRequestDBEntry::CRewardRequestDBEntry(
+    const std::set<CRewardRequest> & p_requests
+)
+{
+    SetNull();
+
+    for (auto const & currReq : p_requests) {
+        requests.insert(currReq);
+    }
+}
+
+CRewardRequestDB::CRewardRequestDB(size_t nCacheSize, bool fMemory, bool fWipe) : CDBWrapper(GetDataDir() / "rewardrequest", nCacheSize, fMemory, fWipe) {
```

Comment:
I think we should wrap all rewards databases into the folder rewards
``` GetDataDir() / "rewards/rewardrequest"```

Reply:
Done